web/src/components/NearbyCities.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { ReactElement } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+
+import { getNearbyCities, LocationType } from 'shared'
+import { CityModel } from 'shared/api'
+
+import CityEntry from './CityEntry'
+import { CityListParent } from './CitySelector'
+
+const NearbyMessageContainer = styled.div`
+  padding: 7px;
+  flex-direction: row;
+  justify-content: space-between;
+`
+
+const NearbyMessage = styled.span`
+  color: ${props => props.theme.colors.textColor};
+  font-family: ${props => props.theme.fonts.web.decorativeFont};
+  padding-top: 15px;
+`
+
+type NearbyCitiesProps = {
+  userLocation: LocationType | null
+  cities: CityModel[]
+  language: string
+  filterText: string
+}
+
+const NearbyCities = ({ userLocation, cities, language, filterText }: NearbyCitiesProps): ReactElement => {
+  const { t } = useTranslation('landing')
+  const nearbyCities = userLocation
+    ? getNearbyCities(
+        userLocation,
+        cities.filter(city => city.live),
+      )
+    : []
+
+  return nearbyCities.length > 0 ? (
+    <div key='nearbyCities'>
+      <CityListParent $stickyTop={0}>{t('nearbyCities')}</CityListParent>
+      {nearbyCities.map(city => (
+        <CityEntry key={city.code} city={city} language={language} filterText={filterText} />
+      ))}
+    </div>
+  ) : (
+    <NearbyMessageContainer>
+      <NearbyMessage>{userLocation ? t('noNearbyCities') : t('locationError')}</NearbyMessage>
```

Comment:
:wrench: I think the message should also be wrapped in the `CityListParent` such that the context is clear. Also, it looks better imo. Not sure if we should not also display more specific error messages and need a retry button like in native. It might happen that location is allowed, but not found.

Reply:
There is a `nearbyCities.length > 0` check and if it's not null it means enabled.. it should show noNearbyCities.
 I will add a refresh button

Reply:
Please have a look how it is implemented on native. There is two things to my comment:
- The text and separator `Nearby Cities` should still be displayed, even if we show an error message and not actual nearby cities
- The refresh button should be shown to try to load the position if it was not successful initially